tools/gen_fresh_dirinfo.py

```diff
@@ -58,8 +67,24 @@
 FLAG_GUARD = "Guard"
 FLAG_STABLE = "Stable"
 
+# Names of the directory authorities.
+AUTHORITIES_NAMES = (
+    "moria1",
+    "tor26",
+    "dizum",
+    "gabelmoo",
+    "dannenberg",
+    "maatuska",
+    "Faravahar",
+    "longclaw",
+    "bastet",
+)
+
+# Authority we trust to measure the mean time before failure (MTBF).
+AUTHORITY_MTBF_MEASURE = "moria1"
```

Comment:
Same question as above: why do you trust `moria1`?

Reply:
Good point. I am going to add more info there.

If you want the full story: We implicitly trust that a majority of the directory authorities will provide accurate information when we use the consensus. For the publication of the MTBF measurement, it is a bit more difficult, because this is a recent feature of Tor, and not all directory authorities run with the latest version of Tor. From what I know _moria1_ and _maatuska_ are currently the only 2 directory authorities which publish these stats, and they are currently not published in the consensus, so we have to rely on the vote of an authority.

We can also increase the reliability of these measurements by downloading the consensus of these 2 authorities, and use the average of the two MTBF measurements, but that will increase the complexity of the code, so I am not sure it is worthwhile to do it.

Reply:
That's a good explanation to put in the comment, together with a date:

```
/// As of July 2021, MTBF measurement ...
```